onnxruntime/core/framework/iexecutor.h

```diff
@@ -34,15 +40,22 @@ class IExecutor {
                          std::vector<OrtValue>& fetches,
                          const logging::Logger& logger) {
     std::unordered_map<size_t, CustomAllocator> fetch_allocators;
-    return Execute(session_state, feed_mlvalue_idxs, feeds, fetch_mlvalue_idxs, fetches, fetch_allocators, logger);
+    return Execute(session_state, feed_mlvalue_idxs, feeds, fetch_mlvalue_idxs, fetches, fetch_allocators, logger, nullptr);
   }
 
   // TODO: as fetch_allocators is optional, it should be a pointer instead of reference
-  virtual common::Status Execute(const SessionState& session_state, const std::vector<int>& feed_mlvalue_idxs,
-                                 const std::vector<OrtValue>& feeds, const std::vector<int>& fetch_mlvalue_idxs,
+  virtual common::Status Execute(const SessionState& session_state,
+                                 const std::vector<int>& feed_mlvalue_idxs,
+                                 const std::vector<OrtValue>& feeds,
+                                 const std::vector<int>& fetch_mlvalue_idxs,
                                  std::vector<OrtValue>& fetches,
                                  // optional custom allocators. key is index in fetches
                                  const std::unordered_map<size_t, CustomAllocator>& fetch_allocators,
-                                 const logging::Logger& logger) = 0;
+                                 const logging::Logger& logger,
+                                 const AllocatorPtr custom_cpu_allocator) = 0;
+                                 
+ protected:
+  const bool& terminate_flag_;
+  const std::unordered_map<string, void*> provider_run_options_;
 };
```

Comment:
Why is it necessary to pull these up to the IExecutor level?

Also avoid protected members. [https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#c133-avoid-protected-data](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#c133-avoid-protected-data)

Reply:
Resolved in IT2

Reply:
> @orereo can you resolve conflicts if this PR is still relevant?

Ack, ETA next week.